inc/templates/head-management.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Automatically inject an `irving/head` component to manage the document
+ * <head>.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\Templates;
+
+use WP_Irving\Components\Component;
+use WP_Irving\Utils;
+
+/**
+ * Manage the <head> by automatically inserting an `irving/head` component.
+ *
+ * @param array     $data    Data object to be hydrated by templates.
+ * @param \WP_Query $query   The current WP_Query object.
+ * @param string    $context The context for this request.
+ * @return array The updated endpoint data.
+ */
+function setup_head(
+	array $data,
+	\WP_Query $query,
+	string $context
+): array {
+
+	// Disable `irving/head` management via filter.
+	if ( ! apply_filters( 'wp_irving_setup_head', true ) ) {
+		return $data;
+	}
+
+	// Unshift a `irving/head` component to the top of the `defaults` array.
+	if ( 'site' === $context ) {
+		array_unshift(
+			$data['defaults'],
+			new Component(
+				'irving/head',
+				[
+					'config' => [
+						'context' => 'defaults',
+					],
+				]
+			)
+		);
+	}
+
+	// Unshift a `irving/head` component to the top of the `page` array.
+	array_unshift(
+		$data['page'],
+		new Component(
+			'irving/head',
+			[
+				'config' => [
+					'context' => 'page',
+				],
+			]
+		)
+	);
+
+	return $data;
+}
+
+/**
+ * Capture the markup output by WordPress for the favicon.
+ *
+ * @return string
+ */
+function get_favicon_markup(): string {
+	ob_start();
+	wp_site_icon();
+	return trim( ob_get_clean() );
+}
+
+/**
+ * Parse WP's favicon markup and inject it into the `irving/head` component.
+ *
+ * @param array $children Children for the <head>.
+ * @return array
+ */
+function inject_favicon( array $children ): array {
+	return array_merge(
+		$children,
+		\WP_Irving\Utils\html_to_components( get_favicon_markup(), [ 'link', 'meta' ] )
+	);
+}
+add_action( 'wp_irving_defaults_head_component_children', __NAMESPACE__ . '\inject_favicon' );
```

Comment:
Move this `add_action()` call into a `bootstrap()` function at the top of the file, and then call the bootstrap function after this file is loaded. See: https://github.com/alleyinteractive/wp-irving/blob/master/wp-irving.php#L77-L78